server/src/main/java/org/elasticsearch/transport/SniffConnectionStrategy.java

```diff
@@ -242,7 +242,7 @@ private void collectRemoteNodes(Iterator<Supplier<DiscoveryNode>> seedNodesSuppl
                     }
                 }
                 logger.warn(() -> "fetching nodes from external cluster [" + clusterAlias + "] failed", e);
-                listener.onFailure(e);
+                listener.onFailure(new NoSeedNodeLeftException(strategyType(), clusterAlias, e));
```

Comment:
For consistency between the two modes, I am also updating this to be wrapped within a `NoSeedNodeLeftException` which makes the error message more explanatory in REST response.

Reply:
I think this only makes sense if `seedNodesSuppliers.hasNext() == false`. If we encountered some other more fundamental error then we didn't run out of seed nodes here.

Also IMO it would be good to eliminate the `seedNodesSuppliers.hasNext() == false` case entirely in this method so that there's always a cause to report.

Reply:
> I think this only makes sense if seedNodesSuppliers.hasNext() == false
Good catch! Will fix.

> Also IMO it would be good to eliminate the `seedNodesSuppliers.hasNext() == false` case entirely in this method so that there's always a cause to report.

Sorry but I don't quite follow. There is no existing `seedNodesSuppliers.hasNext() == false` case. Are you referring to the `else` branch of the openning `if (seedNodesSuppliers.hasNext())`, i.e. related to the above comment? I think in practice the `else` branch never runs. So the cause is always reported. Please let me know if you'd like to drop the `else` branch for both Sniff and Proxy strategies. I can make the change to both. I just want them to be consistent.

Reply:
Yes I meant the `else` branch. I wonder if it could run if `seedNodesSuppliers` is empty. Not sure if this can happen but maybe it could in future even if not today? I'd like to protect against that at least, but again I think it's better to do so up-front rather than on every iteration.